modules/402-ingress-nginx/images/kruise/patches/add-pdb.patch

```diff
@@ -249,3 +266,54 @@ index 1934f145..f5e7bbc1 100644
  func (dsc *ReconcileDaemonSet) storeDaemonSetStatus(ds *appsv1alpha1.DaemonSet, desiredNumberScheduled, currentNumberScheduled, numberMisscheduled, numberReady, updatedNumberScheduled, numberAvailable, numberUnavailable int, updateObservedGen bool, hash string) error {
  	if int(ds.Status.DesiredNumberScheduled) == desiredNumberScheduled &&
  		int(ds.Status.CurrentNumberScheduled) == currentNumberScheduled &&
+diff --git a/pkg/controller/daemonset/daemonset_event_handler.go b/pkg/controller/daemonset/daemonset_event_handler.go
+index 26a33f3d..68c5ec35 100644
+--- a/pkg/controller/daemonset/daemonset_event_handler.go
++++ b/pkg/controller/daemonset/daemonset_event_handler.go
+@@ -284,6 +284,23 @@ func (e *nodeEventHandler) Update(evt event.UpdateEvent, q workqueue.RateLimitin
+ }
+ 
+ func (e *nodeEventHandler) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
++	dsList := &appsv1alpha1.DaemonSetList{}
++	err := e.reader.List(context.TODO(), dsList)
++	if err != nil {
++		klog.V(4).Infof("Error enqueueing daemon sets: %v", err)
++		return
++	}
++
++	node := evt.Object.(*v1.Node)
++	for i := range dsList.Items {
++		ds := &dsList.Items[i]
++		if shouldSchedule, _ := nodeShouldRunDaemonPod(node, ds); shouldSchedule {
++			q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
++				Name:      ds.GetName(),
++				Namespace: ds.GetNamespace(),
++			}})
++		}
++	}
+ }
+ 
```

Comment:
reconcile daemonsets if a node has been deleted, for the sake of filling the daemonset controller in on the latest changes